changelog/922.improvement.md

```diff
@@ -0,0 +1,31 @@
+## Problem
+Rasa knowledge base actions cannot infer the object type of an object directly from the user message. This prevents action_query_knowledge from providing a suitable response when a user asks for a certain attribute of an object unless the user first asks to list objects related to that object type even though the knowledge base has the relevant information. That is, the knowledge base actions require the slot `object_type` to be set to one of the primary key values in the knowledge base for it to search through the objects. Here is an example:
```

Comment:
I think we can frame the problem better. Especially the first sentence seems like Rasa knowledge base actions cannot infer the object type at all. And words like "suitable response" are pretty vague better if we can clarify exactly. For example,

_Rasa knowledge base action cannot currently infer the object type of an object when the user only references the object in the message. For example <<insert example quote message>>_.....

Reply:
In addition the second sentence seems too long and complex, better if we can simplify it using a tool like [grammarly](grammarly.com) for example